examples/favorites_advanced/getx_favorites_advanced/lib/feature_puppy/controllers/puppy_list_controller.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+
+import 'package:favorites_advanced_base/repositories.dart';
+import 'package:favorites_advanced_base/models.dart';
+
+class PuppyListController extends GetxController {
+  PuppyListController(){
+    _repository = Get.find<PuppiesRepository>();
+    _initPuppies();
+  }
+
+  PuppiesRepository _repository;
+  final _puppies = [].obs;
+  final isLoading = false.obs;
+  final hasError = false.obs;
+
+  List<Puppy> get searchedPuppies => [..._puppies];
+
+  Future<List<Puppy>> filterPuppies({String query}) async {
+    final puppies = await _repository.getPuppies(query: query);
+    return puppies;
+  }
+
+  Future<void> _initPuppies() async {
+    _showLoading();
+    try{
+      // await Future.delayed(const Duration(seconds: 1));
+    final puppies = await _repository.getPuppies();
+    _puppies.assignAll(puppies);
+    // throw Exception('Error');
+    }catch(e){
+      _setError();
+      print(e.toString());
+    }
+    _hideLoading();
+  }
+
+  void _showLoading(){
```

Comment:
@RomanovaPrime don't you have any warning on your IDE to convert this to expression body?

Reply:
Actually not, but I refactored the code